src/hotspot/share/memory/metaspaceShared.cpp

```diff
@@ -122,7 +123,27 @@ char* MetaspaceShared::symbol_space_alloc(size_t num_bytes) {
   return _symbol_region.allocate(num_bytes);
 }
 
-size_t MetaspaceShared::reserved_space_alignment() { return os::vm_allocation_granularity(); }
+void MetaspaceShared::init_alignments() {
+  assert(_core_region_alignment == 0, "call this only once");
+  _core_region_alignment = (size_t)os::vm_allocation_granularity();
+  if (DumpSharedSpaces) {
+    // os::vm_allocation_granularity() is usually 4K for most OSes. However, on Linux/aarch64,
+    // it can be either 4K or 64K. To generate archives that are compatible for
+    // both settings, always dump the static archive with 64K alignment.
+    // The dynamic archive will always use the same value as the base archive.
+    if (_core_region_alignment < 64*K) {
+      log_info(cds)("Set core region alignment to 64K for all platforms");
+      _core_region_alignment = 64*K;
+    }
+  } else {
+    assert(UseSharedSpaces, "don't call this if not mapping at least the base archive");
+    assert(FileMapInfo::current_info() != NULL, "Call init_alignments() after base archive is opened");
+    _core_region_alignment = FileMapInfo::current_info()->region_alignment();
+    assert(_core_region_alignment == 64*K, "CDS always use 64K region alignment");
```

Comment:
This should be `_core_region_alignment >= 64*K, "CDS must use minimum 64K region alignment"`. and moved below the `else` block.

This makes the code compatible with future OSes that may use >64KB for `os::vm_allocation_granularity()`.